mlflow/genai/scorers/deepeval/registry.py

```diff
@@ -0,0 +1,81 @@
+
+from mlflow.exceptions import MlflowException
+from mlflow.genai.scorers.deepeval.utils import DEEPEVAL_NOT_INSTALLED_ERROR_MESSAGE
+from mlflow.protos.databricks_pb2 import BAD_REQUEST
+
+_METRIC_REGISTRY = {
+    # RAG Metrics
+    "AnswerRelevancy": "deepeval.metrics.AnswerRelevancyMetric",
+    "Faithfulness": "deepeval.metrics.FaithfulnessMetric",
+    "ContextualRecall": "deepeval.metrics.ContextualRecallMetric",
+    "ContextualPrecision": "deepeval.metrics.ContextualPrecisionMetric",
+    "ContextualRelevancy": "deepeval.metrics.ContextualRelevancyMetric",
+    # Agentic Metrics
+    "TaskCompletion": "deepeval.metrics.TaskCompletionMetric",
+    "ToolCorrectness": "deepeval.metrics.ToolCorrectnessMetric",
+    "ArgumentCorrectness": "deepeval.metrics.ArgumentCorrectnessMetric",
+    "StepEfficiency": "deepeval.metrics.StepEfficiencyMetric",
+    "PlanAdherence": "deepeval.metrics.PlanAdherenceMetric",
+    "PlanQuality": "deepeval.metrics.PlanQualityMetric",
+    # Conversational Metrics
+    # TODO: Add support for conversational metrics with ConversationalTestCase
+    # "TurnRelevancy": "deepeval.metrics.TurnRelevancyMetric",
+    # "RoleAdherence": "deepeval.metrics.RoleAdherenceMetric",
+    # "KnowledgeRetention": "deepeval.metrics.KnowledgeRetentionMetric",
+    # "ConversationCompleteness": "deepeval.metrics.ConversationCompletenessMetric",
+    # "GoalAccuracy": "deepeval.metrics.GoalAccuracyMetric",
+    # "ToolUse": "deepeval.metrics.ToolUseMetric",
+    # "TopicAdherence": "deepeval.metrics.TopicAdherenceMetric",
+    # Safety Metrics
+    "Bias": "deepeval.metrics.BiasMetric",
+    "Toxicity": "deepeval.metrics.ToxicityMetric",
+    "NonAdvice": "deepeval.metrics.NonAdviceMetric",
+    "Misuse": "deepeval.metrics.MisuseMetric",
+    "PIILeakage": "deepeval.metrics.PIILeakageMetric",
+    "RoleViolation": "deepeval.metrics.RoleViolationMetric",
+    # General Metrics
+    "Hallucination": "deepeval.metrics.HallucinationMetric",
+    "Summarization": "deepeval.metrics.SummarizationMetric",
+    "JsonCorrectness": "deepeval.metrics.JsonCorrectnessMetric",
+    "PromptAlignment": "deepeval.metrics.PromptAlignmentMetric",
+    # Deterministic Metrics
+    "ExactMatch": "deepeval.metrics.ExactMatchMetric",
+    "PatternMatch": "deepeval.metrics.PatternMatchMetric",
+}
+
+
+def get_metric_class(metric_name: str):
+    """
+    Get DeepEval metric class by name.
+
+    Args:
+        metric_name: Name of the metric (e.g., "AnswerRelevancy", "Faithfulness")
+
+    Returns:
+        The DeepEval metric class
+
+    Raises:
+        MlflowException: If the metric name is not recognized or deepeval is not installed
+    """
+    if metric_name not in _METRIC_REGISTRY:
+        available_metrics = ", ".join(sorted(_METRIC_REGISTRY.keys()))
+        raise MlflowException(
+            f"Unknown metric: '{metric_name}'. Available metrics: {available_metrics}",
+            error_code=BAD_REQUEST,
+        )
```

Comment:
```suggestion
        raise MlflowException.invalid_parameter_value(
            f"Unknown metric: '{metric_name}'. Available metrics: {available_metrics}",
        )
```
Let's avoid raising BAD_REQUEST for invalid values error